pkg/controller/azvolume.go

```diff
@@ -140,7 +140,26 @@ func (r *reconcileAzVolume) TriggerDelete(ctx context.Context, volumeName string
 	if err := r.UpdateStatus(ctx, azVolume.Name, true); err != nil {
 		return err
 	}
-	klog.Infof("successfully deleted volume (%s)and update status of AzVolume (%s)", azVolume.Spec.UnderlyingVolume, azVolume.Name)
+
+	// Delete all AzVolumeAttachment objects bound to the deleted AzVolume
```

Comment:
Given that there's a finalizer for AzVolumeAttachment on the AzVolume object, can there be a scenario where AzVolume gets deleted while the corresponding AzVolumeAttachment still exists?

If not, then do we even need to handle this?

Reply:
I was considering any unlikely scenario where for whatever reason there is an uncleaned up AzVolumeAttachment objects for AzVolume required to be deleted (not sure if this would be a possible scenario but thought it was worth venturing). If there is no trigger to clean up these AzVolumeAttachment objects, then both the AzVolume and its AzVolumeAttachment objects will be be left undeleted.